deneb.go

```diff
@@ -0,0 +1,239 @@
+			Amount:         phase0.Gwei(withdrawal.Amount),
+			Address:        b20(withdrawal.Address),
+		})
+	}
+
+	// BlobsBundle
+	blobsBundle := &apiDeneb.BlobsBundle{}
+	for _, commitment := range block.BlobsBundle.Commitments {
+		// Unmarshal commitment
+		var commitmentObj consensus.KZGCommitment
+		err := commitmentObj.UnmarshalJSON(commitment)
+		if err != nil {
+			panic(err)
+		}
+		blobsBundle.Commitments = append(blobsBundle.Commitments, consensus.KZGCommitment(commitmentObj))
+	}
+
+	for _, proof := range block.BlobsBundle.Proofs {
+		// Unmarshal proof
+		var proofObj consensus.KZGProof
+		err := proofObj.UnmarshalJSON(proof)
+		if err != nil {
+			panic(err)
+		}
+		blobsBundle.Proofs = append(blobsBundle.Proofs, consensus.KZGProof(proofObj))
+	}
+
+	for _, blob := range block.BlobsBundle.Blobs {
+		// Unmarshal data
+		var blobObj consensus.Blob
+		err := blobObj.UnmarshalJSON(blob)
+		if err != nil {
+			panic(err)
+		}
+		blobsBundle.Blobs = append(blobsBundle.Blobs, consensus.Blob(blobObj))
+	}
+
+	return &apiDeneb.SubmitBlockRequest{
+		Message: &v1.BidTrace{
+			Slot:                 block.BidTrace.Slot,
+			ParentHash:           b32(block.BidTrace.ParentHash),
+			BlockHash:            b32(block.BidTrace.BlockHash),
+			BuilderPubkey:        b48(block.BidTrace.BuilderPubkey),
+			ProposerPubkey:       b48(block.BidTrace.ProposerPubkey),
+			ProposerFeeRecipient: b20(block.BidTrace.ProposerFeeRecipient),
+			GasLimit:             block.BidTrace.GasLimit,
+			GasUsed:              block.BidTrace.GasUsed,
+			Value:                uint256.MustFromHex(block.BidTrace.Value),
+		},
+		ExecutionPayload: &consensus.ExecutionPayload{
+			ParentHash:    b32(block.ExecutionPayload.ParentHash),
+			StateRoot:     b32(block.ExecutionPayload.StateRoot),
+			ReceiptsRoot:  b32(block.ExecutionPayload.ReceiptsRoot),
+			LogsBloom:     b256(block.ExecutionPayload.LogsBloom),
+			PrevRandao:    b32(block.ExecutionPayload.PrevRandao),
+			BaseFeePerGas: byteSliceToUint256Int(block.ExecutionPayload.BaseFeePerGas),
+			FeeRecipient:  b20(block.ExecutionPayload.FeeRecipient),
+			BlockHash:     b32(block.ExecutionPayload.BlockHash),
+			ExtraData:     block.ExecutionPayload.ExtraData,
+			BlockNumber:   block.ExecutionPayload.BlockNumber,
+			GasLimit:      block.ExecutionPayload.GasLimit,
+			Timestamp:     block.ExecutionPayload.Timestamp,
+			GasUsed:       block.ExecutionPayload.GasUsed,
+			Transactions:  transactions,
+			Withdrawals:   withdrawals,
+			BlobGasUsed:   block.ExecutionPayload.BlobGasUsed,
+			ExcessBlobGas: block.ExecutionPayload.ExcessBlobGas,
+		},
+		BlobsBundle: blobsBundle,
+		Signature:   b96(block.Signature),
+	}
+}
+
+// Add Commitments, Proofs, Data to BlobsBundle
+func convertBlobBundleToProto(blobBundle *apiDeneb.BlobsBundle) *BlobsBundle {
+	protoBlobsBundle := &BlobsBundle{}
+
+	for _, commitment := range blobBundle.Commitments {
+		// Marshal commitment
+		commitmentBytes, err := commitment.MarshalJSON()
```

Comment:
should we just use the raw bytes rather than marshaling to/from JSON?

Reply:
Better to define as much as the type as possible (we know the internal types are of an array)

Reply:
i just mean `commitment` is already a `[48]byte` so we can just copy the bytes instead of calling `MarshalJSON()`

Reply:
Fixed